### DISCUSSION TITLE
How to show activity indication where we dont know about the current progress ?

### DISCUSSION THREAD

**Discussion by heejit** [cogentcore/core#1235] [General]
I want to show user that some activity is going on but I don't have any progress value to update the meter is there any other widget or any other there way to use meter.

**Comment by kkoreilly** [comment]
You could use a goroutine to make the meter quickly go from 0 to 1 and then reset and repeat to indicate uncertain progress.

**Reply by heejit** [reply]
I try following but meter updated only after everything completed 
```
        this.ProgressMeter = core.NewMeter(vFrame2)
	this.ProgressMeter.SetMin(1).SetMax(100)

	vTicker = time.NewTicker(500 * time.Millisecond)
	progressUpdate := func() {
		for range vTicker.C {
			if this.ProgressMeter.Value >= 100 {
				this.ProgressMeter.SetValue(1)
			} else {
				this.ProgressMeter.SetValue(this.ProgressMeter.Value + 50)
			}
			this.ProgressMeter.UpdateRender()
			fmt.Println("updated meter")
		}
	}
	go progressUpdate()

```

**Reply by kkoreilly** [reply]
You need to use `AsyncLock` and `AsyncUnlock`; see https://www.cogentcore.org/core/basics/async

**Reply by heejit** [reply]
I try this not working it shows that timer run only two times and text updated only after 
process completed  and process took 6 seconds to finish is that a issue is not getting updated
or am i missing something.

```
       vTicker = time.NewTicker(100 * time.Millisecond)
	progressUpdate := func() {
		i := 1
		for range vTicker.C {
			v := fmt.Sprintf("Working : %d", i)
			this.ProgressText.AsyncLock()
			this.ProgressText.SetText(v)
			this.ProgressText.Update()
			this.ProgressText.AsyncUnlock()
			fmt.Println("Progress : ", v)
			i += 1
		}
	}
	go progressUpdate()       
```

**Reply by heejit** [reply]
Sometime process took 12 sec also but timer shows run only two times

**Reply by heejit** [reply]
If I remove the asynclock and unlock timer runs more than 2 times but text is not getting updated during the process it always after process finished.

**Reply by kkoreilly** [reply]
Are you running the process itself in the event loop? You also need to run whatever action you are doing in a goroutine so it doesn't block the event loop thread.

**Reply by heejit** [reply]
I am running the function on click of button then inside the click handler i am calling another function to get result from database then display the result.

**Reply by heejit** [reply]
It works when I run the click handler in goroutine

thanks.
